src/ring_theory/ideal/norm.lean

```diff
@@ -240,3 +243,169 @@ noncomputable def ideal.abs_norm [infinite S] [is_dedekind_domain S]
   map_mul' := λ I J, by rw card_quot_mul,
   map_one' := by rw [ideal.one_eq_top, card_quot_top],
   map_zero' := by rw [ideal.zero_eq_bot, card_quot_bot] }
+
+namespace ideal
+
+variables [infinite S] [is_dedekind_domain S] [module.free ℤ S] [module.finite ℤ S]
+
+lemma abs_norm_apply (I : ideal S) : abs_norm I = card_quot I := rfl
+
+@[simp] lemma abs_norm_bot : abs_norm (⊥ : ideal S) = 0 :=
+by rw [← ideal.zero_eq_bot, _root_.map_zero]
+
+@[simp] lemma abs_norm_top : abs_norm (⊤ : ideal S) = 1 :=
+by rw [← ideal.one_eq_top, _root_.map_one]
+
+@[simp] lemma abs_norm_eq_one_iff {I : ideal S} : abs_norm I = 1 ↔ I = ⊤ :=
+by rw [abs_norm_apply, card_quot_eq_one_iff]
+
+/-- Let `e : S ≃ I` be an additive homomorphism (therefore a `ℤ`-linear equiv).
+Then an alternative way to compute the norm of `I` is given by taking the determinant of `e`. -/
```

Comment:
```suggestion
Then an alternative way to compute the norm of `I` is given by taking the determinant of `e`. See `nat_abs_det_basis_change` for a more familiar formulation of this result. -/
```
This is maybe subjective, but I think that number theorists know that the norm is the determinant of the base change matrix, but hardly think in term of determinant of linear isomorphisms here.

Reply:
```suggestion
Then an alternative way to compute the norm of `I` is given by taking the determinant of `e`.
See `nat_abs_det_basis_change` for a more familiar formulation of this result. -/
```